src/main/kotlin/no/nav/familie/baks/soknad/api/controllers/SøknadController.kt

```diff
@@ -27,27 +26,133 @@ class SøknadController(
     private val kontantstøtteSøknadService: KontantstøtteSøknadService,
     private val barnetrygdSøknadService: BarnetrygdSøknadService
 ) {
-    @Deprecated("Bruk nytt endepunkt med oppdatert kontrakt /soknad/v9 ")
-    @PostMapping("/soknad/v8")
-    fun søknadsmottakBarnetrygd(
-        @RequestBody(required = true) søknad: BarnetrygdSøknadV8
-    ): ResponseEntity<Ressurs<Kvittering>> = ResponseEntity.ok().body(barnetrygdSøknadService.mottaOgSendBarnetrygdsøknad(søknad))
-
     @PostMapping("/soknad/v9")
     fun søknadsmottakBarnetrygd(
         @RequestBody(required = true) søknad: BarnetrygdSøknadV9
-    ): ResponseEntity<Ressurs<Kvittering>> = ResponseEntity.ok().body(barnetrygdSøknadService.mottaOgSendBarnetrygdsøknad(søknad))
-
-    @Deprecated("Bruk nytt endepunkt med oppdatert kontrakt /soknad/kontantstotte/v5 ")
-    @PostMapping("/soknad/kontantstotte/v4")
-    fun søknadsmottakKontantstøtte(
-        @RequestBody(required = true)
-        kontantstøtteSøknad: KontantstøtteSøknadV4
-    ): ResponseEntity<Ressurs<Kvittering>> = ResponseEntity.ok().body(kontantstøtteSøknadService.mottaOgSendKontantstøttesøknad(kontantstøtteSøknad))
+    ): ResponseEntity<Ressurs<Kvittering>> {
+        søknad.valider()
+        return ResponseEntity.ok().body(barnetrygdSøknadService.mottaOgSendBarnetrygdsøknad(søknad))
+    }
 
     @PostMapping("/soknad/kontantstotte/v5")
     fun søknadsmottakKontantstøtte(
         @RequestBody(required = true)
         kontantstøtteSøknad: KontantstøtteSøknadV5
-    ): ResponseEntity<Ressurs<Kvittering>> = ResponseEntity.ok().body(kontantstøtteSøknadService.mottaOgSendKontantstøttesøknad(kontantstøtteSøknad))
+    ): ResponseEntity<Ressurs<Kvittering>> {
+        kontantstøtteSøknad.valider()
+        return ResponseEntity.ok().body(kontantstøtteSøknadService.mottaOgSendKontantstøttesøknad(kontantstøtteSøknad))
+    }
+}
+
+fun BarnetrygdSøknadV9.valider() {
+    // valider ident (forhindre SQL/NoSQL injection)
+    søker.ident.verdi.values.forEach { fnr ->
+        val erGyldig = runCatching { Fødselsnummer(fnr) }.isSuccess
+        require(erGyldig) { "Ugyldig format på fødselsnummer" }
+    }
+
+    barn.forEach { barn ->
+        barn.ident.verdi.values.forEach { fnr ->
+            val erGyldig = runCatching { Fødselsnummer(fnr) }.isSuccess
+            require(erGyldig || fnr.all { it.isDigit() }) { "Ugyldig format på barnets fødselsnummer" }
+        }
+    }
+
+    // XSS prevention - sanitize text fields
+    listOfNotNull(
+        søker.navn,
+        søker.statsborgerskap,
+        søker.sivilstand,
+        søker.adresse,
+        søker.nåværendeSamboer
+    ).forEach { textField ->
+        // valider alle verdier i tekstfelt
+        textField.verdi.values.forEach { verdi ->
+            require(verdi.toString().length < 500) { "Tekstfelt er for langt" }
```

Comment:
Er det noe validering i frontend på lengder? 500 er ganske langt for mange av disse feltene vil jeg tro

Reply:
kortet ned på lengden, tror ikke det er noen validering i frontend.